### PR TITLE
Added Basic Pic Of the Day App Widget

### LINKED ISSUE #1296: Widget
**Summary:** 
It would be great if there is a widget that shows the picture of the day from [website](https://commons.wikimedia.org/wiki/Main_Page) [Edited by misaochan: pic of the day has been implemented, but a more full-featured widget would be awesome!]

screenshot-commons wikimedia org-2018 03 13-10-20-18

### PR DESCRIPTION
**Pull request by madhurgupta10**
## Description
Added Basic Pic of the Day App Widget

Fixes
#1296 

Also fixed indentation using Android Studio's AutoIndent feature.

## Tests performed
Tested on Android 8.0 Oreo on Xiaomi Mi A1.

## Screenshots showing what changed (optional)
screenshot_20180516-231353

### CODE REVIEW

**Comment by madhurgupta10** [comment]
Hi @maskaravivek the build is failing because I have added an external jar file following this [link](https://stackoverflow.com/questions/26676232/convert-xml-to-json-cant-find-working-methodxml-tojsonobject-doesnt-work)

**Comment by maskaravivek** [comment]
Why do you need to convert XML to JSON? You can parse the XML response itself. If you are not comfortable parsing the XML, use the JSON version of the API.

File: app/build.gradle [hunk 1]
```diff
@@ -129,6 +121,7 @@ android {
     flavorDimensions 'tier'
     productFlavors {
         prod {
+            buildConfigField "String", "WIKIMEDIA_API_POTD", "\"https://commons.wikimedia.org/w/api.php?action=featuredfeed&feed=potd&feedformat=rss&language=en\""
```
[the remark this answers is not in the document]
Comment (misaochan) [reply app/build.gradle]: Interesting, that usually works! Could be worth asking on wikitech-l or the Commons village pump?

[the remark this answers is not in the document]
Comment (madhurgupta10) [reply app/build.gradle]: I don't know, what I can do is that I stick to XML and use XMLPullParser for parsing without converting it to json

**Comment by maskaravivek** [comment]
@madhurgupta10 Yes, I checked that the feed you are using always returns an XML response. 

I found an alternate API that can be used to get the picture of the day. The results returned by both the APIs match. 

https://commons.wikimedia.org/w/api.php?action=query&generator=images&titles=Template:Potd/2018-05-17&prop=imageinfo&iiprop=url|size|user|mime|extmetadata

This will return a JSON response.

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -23,81 +24,75 @@
...
-
         <activity
             android:name=".nearby.NearbyActivity"
             android:label="@string/title_activity_nearby"
             android:parentActivityName=".contributions.ContributionsActivity" />
-
         <activity
             android:name=".notification.NotificationActivity"
             android:label="@string/navigation_item_notification" />
-
```
Comment (maskaravivek) [inline app/src/main/AndroidManifest.xml]: Please undo all such formatting changes that don't apply to this diff.

Comment (madhurgupta10) [reply app/src/main/AndroidManifest.xml]: @maskaravivek Sure, I will undo all such changes, sorry for a bad quality pull request, I am new to open source :disappointed:

Comment (misaochan) [reply app/src/main/AndroidManifest.xml]: Don't feel bad. Most of us need to make changes to our PRs before they are accepted. ;)

Comment (maskaravivek) [reply app/src/main/AndroidManifest.xml]: Yup @madhurgupta10 you have really done a great job with this PR. I personally wanted this feature in the app as soon as possible and with your PR we are very close to it. :)

Comment (madhurgupta10) [reply app/src/main/AndroidManifest.xml]: @maskaravivek @misaochan Thanks, I will update the PR asap.

File: app/build.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (nicolas-raoul) [reply app/build.gradle]: The API documentation at https://commons.wikimedia.org/w/api.php?action=help&modules=featuredfeed says `feedformat` must be `One of the following values: rss, atom`.

**Comment by maskaravivek** [comment]
@madhurgupta10 Also extract the logic of getting the picture of the day outside of the widget class, so that it can be reused.

**Comment by madhurgupta10** [comment]
@maskaravivek how to parse page numbers from the json api?
image
image

**Comment by maskaravivek** [comment]
The best way would be to take the content inside `pages` as a `Map<String, Page>`

**Comment by nicolas-raoul** [comment]
If all you need is the first element, do you really need to parse page numbers or even save pages?

File: app/build.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (madhurgupta10) [reply app/build.gradle]: Hi, I removed the XML to JSON conversion, instead, I used JSoup to simply get the src attribute from img tag.

**Comment by madhurgupta10** [comment]
Hi, I made the necessary changes but after resolving the merge conflict the build seems to be failing

**Comment by maskaravivek** [comment]
This is the error @madhurgupta10 screen shot 2018-06-02 at 9 52 13 am

Try rebasing your branch with master once.

**Comment by maskaravivek** [comment]
@madhurgupta10 I just pulled your branch. I couldn't figure out how to add the widget. Can you elaborate on the steps.

**Comment by madhurgupta10** [comment]
@maskaravivek
ezgif com-optimize

**Comment by madhurgupta10** [comment]
@maskaravivek Hi, if there is any improvement needed, please tell me I will be happy do it 😄

**Comment by madhurgupta10** [comment]
@maskaravivek Thanks for the feedback, I will keep on adding more functionality to the app widget.

**Comment by maskaravivek** [comment]
@misaochan @neslihanturan This PR is ready to be merged. Can this be merged to master or are we waiting for the release to be completed?

**Comment by neslihanturan** [comment]
I think we have to wait for release for new features, however we have a separate release branch, so merging to master shouldn't be a problem. Anyway better to wait for @misaochan 's answer:)

**Comment by misaochan** [comment]
I'm good with merging to master, as we have a release branch. :)

**Pull request merged**